src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionService.kt

```diff
@@ -42,21 +54,83 @@ class IdentificationSessionService(private val identificationSessionRepository:
         return "${applicationProperties.baseUrl}$IDENTIFICATION_SESSIONS_BASE_PATH/${session.useIdSessionId}/$TCTOKEN_PATH_SUFFIX"
     }
 
+    @Transactional
     fun startSessionWithEIdServer(useIdSessionId: UUID): TCTokenType {
-        val identificationSession = identificationSessionRepository.findByUseIdSessionId(useIdSessionId)
-            ?: throw IdentificationSessionNotFoundException(useIdSessionId)
-        val tcToken: TCTokenType
+        val requestData: RequestData = getRequestData(useIdSessionId)
+
+        val tcTokenWrapper: TcTokenWrapper?
         try {
-            val eidService = EidService(eidServiceConfig, identificationSession.requestDataGroups)
-            tcToken = eidService.getTcToken("${applicationProperties.baseUrl}$REFRESH_PATH")
+            tcTokenWrapper = soapHandler.getTcToken(requestData, "${applicationProperties.baseUrl}$REFRESH_PATH")
             tcTokenCallsSuccessfulCounter.increment()
         } catch (e: Exception) {
             tcTokenCallsWithErrorsCounter.increment()
             log.error("Failed to get tc token for identification session. useIdSessionId=$useIdSessionId", e)
             throw e
         }
-        updateEIdSessionId(useIdSessionId, extractEIdSessionId(tcToken))
-        return tcToken
+
+        val session = identificationSessionRepository.findByUseIdSessionId(useIdSessionId)
+        session!!.eIdSessionId = extractEIdSessionId(tcTokenWrapper.tcToken)
```

Comment:
The actual update is happening through the transactional, is that correct? Why do we want to get rid of the update function? Because this is now the only place where it is needed? 
Personally, I think having the function call helps a lot with readability. wdyt?